### PR TITLE
Adding ability to have different tld's

### PR DESCRIPTION
**Pull request by denmat**
- Everything is set with a tld of *.dev
- This allows us to be a little less prescriptive
- Replaced hardcoded tld in templates and configs
- added tests for different tld

This is complementary to https://github.com/boxen/puppet-dnsmasq/pull/13 and helps with https://github.com/boxen/puppet-dnsmasq/issues/16

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -15,7 +15,7 @@
   $logfile    = undef,
 ) {
   require homebrew
-  $servicename = 'dev.dnsmasq'
+  $servicename = "${tld}.dnsmasq"
```
Comment (dgoodlad) [inline manifests/init.pp]: A few of boxen's tools and commands assume that all of its managed services are in the `dev.` namespace. I'd suggest not changing that, even if it ends up serving DNS for a different TLD.

File: manifests/init.pp [hunk 2]
```diff
@@ -27,8 +27,8 @@
     require => File[$configdir],
   }
 
-  file { '/Library/LaunchDaemons/dev.dnsmasq.plist':
-    content => template('dnsmasq/dev.dnsmasq.plist.erb'),
+  file { "/Library/LaunchDaemons/${tld}.dnsmasq.plist":
```
Comment (dgoodlad) [inline manifests/init.pp]: Same thing here about naming and boxen's internal assumptions.

File: spec/classes/dnsmasq_spec.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper'
 
 describe 'dnsmasq' do
-
+
```
Comment (dgoodlad) [inline spec/classes/dnsmasq_spec.rb]: Whitespace nitpick

File: spec/classes/dnsmasq_spec.rb [hunk 4]
```diff
@@ -25,23 +25,20 @@
...
     should contain_file(configdir).with_ensure('directory')
     should contain_file(datadir).with_ensure('directory')
     should contain_file(logdir).with_ensure('directory')
 
     should contain_file(configfile).with({
-      :content => File.read('spec/fixtures/dnsmasq.conf'),
       :notify  => "Service[#{servicename}]",
       :require => "File[#{configdir}]",
-    })
+    }).with_content(%r{\naddress=/dev/127.0.0.1\nlisten-address=127.0.0.1\n})
```
Comment (dgoodlad) [inline spec/classes/dnsmasq_spec.rb]: :heart:

**Comment by dgoodlad** [comment]
If you fix up the service name, moving it back to `dev.dnsmasq`, this will all look great.

**Comment by denmat** [comment]
@dgoodlad -- this good to merge?

**Comment by denmat** [comment]
@dgoodlad -- bump?

**Comment by jacobbednarz** [comment]
@denmat if you want to resolve these conflicts, i'll get this merged

**Comment by jacobbednarz** [comment]
Thanks! :cake:

**Pull request merged**